### ISSUE TITLE
ZTS: do not depend on installed udev rules / zvol_id

### ISSUE THREAD

**Issue by problame** [openzfs/zfs#12186]
It would be nice to be able to run the ZTS completely in-tree, i.e., without ever running `make install`.

The primary blocker I have encountered is the dependency of tests on /dev/zvol paths which are installed by the udev rules.
If the udev rule and/or `zvol_id` is not installed, these tests fail, e.g.

https://github.com/openzfs/zfs/blob/f645d4416f0f0ddb46ba85bbaaf3bb7e87f75877/tests/zfs-tests/tests/functional/slog/slog_replay_volume.ksh#L62

### How will this feature improve OpenZFS?

It enables developers to check out and run tests on a copy of the source tree without mutating global system state.
(The result directories, disk images for file vdevs, etc, don't count since they are managed by ZTS).
This is useful when working on multiple projects simultaneously on the same machine, e.g., via git worktree.

**Comment by behlendorf** [comment]
You're right, zvols are the big dependency but there are a couple others as well (mount.zfs, vdev_id, etc).  To make developing in-tree easier, and to avoid doing a `make install`, there is a `zfs-helpers.sh` script which can be used.  It still does modify the global system state, but only by adding the bare minimum of support scripts which can't be located anywhere else.  You can also use the script to easily remove those specific files from the system.
